arch/arc/core/thread.c

```diff
@@ -165,6 +165,7 @@ void z_new_thread(struct k_thread *thread, k_thread_stack_t *stack,
 	 */
 	pInitCtx->status32 |= _ARC_V2_STATUS32_US;
 #else /* For no USERSPACE feature */
+	pStackMem += STACK_GUARD_SIZE;
```

Comment:
It would be good to use the macro's from arc/arch.h or even better generic zephyr stack macro's in this file as much as possible. E.g. should this be STACK_GUARD_SIZE,  or Z_ARCH_THREAD_STACK_RESERVED?
Also at other places in this file, e.g. calculation of stack_AdjSize in line 84/86

Reply:
done